### DISCUSSION TITLE
Enable Link-Time Optimization (LTO)

### DISCUSSION THREAD

**Discussion by zamazan4ik** [efugier/smartcat#31] [Ideas]
Hi!

I noticed that in the `Cargo.toml` file Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a bit.

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where additionally to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the application. Adding this option directly to the Cargo.toml file allows users to get an LTO-optimized version if they install the application with `cargo install` command. Using ThinLTO should also help to reduce the build-time overhead with LTO. E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48).

Basically, it can be enabled with the following lines:

```
[profile.release]
lto = true
```

I have made quick tests (Fedora 40) by adding `lto = true` to the Release profile. The binary size reduction is from 7.2 Mib to 5.8 Mib.

Thank you.

**Comment by efugier** [comment]
Good one, it's done starting in [2.0.0](https://github.com/efugier/smartcat/releases/tag/2.0.0) :slightly_smiling_face:
